x-pack/plugins/observability/public/pages/rules/components/last_response_filter.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable react/function-component-definition */
+
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiFilterGroup,
+  EuiPopover,
+  EuiFilterButton,
+  EuiFilterSelectItem,
+  EuiHealth,
+} from '@elastic/eui';
+import { AlertExecutionStatuses, AlertExecutionStatusValues } from '../../../../../alerting/common';
+import { getHealthColor, rulesStatusesTranslationsMapping } from '../config';
+import { StatusFilterProps } from '../types';
+
+export const LastResponseFilter: React.FunctionComponent<StatusFilterProps> = ({
+  selectedStatuses,
+  onChange,
+}: StatusFilterProps) => {
+  const [selectedValues, setSelectedValues] = useState<string[]>(selectedStatuses);
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(selectedValues);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedValues]);
+
+  useEffect(() => {
+    setSelectedValues(selectedStatuses);
+  }, [selectedStatuses]);
+
+  return (
+    <EuiFilterGroup>
+      <EuiPopover
+        isOpen={isPopoverOpen}
+        closePopover={() => setIsPopoverOpen(false)}
+        button={
+          <EuiFilterButton
+            iconType="arrowDown"
+            hasActiveFilters={selectedValues.length > 0}
+            numActiveFilters={selectedValues.length}
+            numFilters={selectedValues.length}
+            onClick={() => setIsPopoverOpen(!isPopoverOpen)}
+            data-test-subj="ruleStatusFilterButton"
+          >
+            <FormattedMessage
+              id="xpack.observability.rules.ruleLastResponseFilterLabel"
+              defaultMessage="Last response"
+            />
+          </EuiFilterButton>
+        }
+      >
+        <div className="euiFilterSelect__items">
+          {[...AlertExecutionStatusValues].sort().map((item: AlertExecutionStatuses) => {
+            const healthColor = getHealthColor(item);
+            return (
+              <EuiFilterSelectItem
+                key={item}
+                style={{ textTransform: 'capitalize' }}
+                onClick={() => {
+                  const isPreviouslyChecked = selectedValues.includes(item);
+                  if (isPreviouslyChecked) {
+                    setSelectedValues(selectedValues.filter((val) => val !== item));
```

Comment:
I mean you can call `onChange` directly

Reply:
I think I know what you mean. If we didn't have the requirement to keep the dropdown open with the selected options marked/checked (this is actually how existing triggres_actions_ui filter works), calling onChange directly would work.